finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.{Buffer, BufferWriter}
+
+trait CanBeParameter[-A] { outer =>
+  /**
+   * Returns the size of the given parameter in its MySQL binary representation.
+   **/
+  def sizeOf(param: A): Int
+
+  /**
+   * Retrieves the MySQL type code for the given parameter.
+   * */
+  def typeCode(param: A): Short
+
+  def write(writer: BufferWriter, param: A): Unit
```

Comment:
I think consistency is better here.  If we're going to change the `write*` style methods, let's do them all at once.  Having some of them which follow one convention, and others that follow another is a bummer.  Also, this matches the way that the `java.nio.Buffer` methods work, so it's not unprecedented.